### ISSUE TITLE
How to config all redirects to offline internal php nginx server

### ISSUE THREAD

**Issue by raspblu** [nodogsplash/nodogsplash#338]
Hello, I am using raspberry pi 3b+ on full Stretch OS. I am trying to configure a setup where anyone nearby is able to connect to the access point of pi (which may or may not have internet) and nodogsplash will pop up and redirect to internal index.php located in /var/www/html/ that is hosted by nginx. I also want any further websites that the user pulls up to also redirect to that index.php. I am able to successfully access the website using localhost, but unable to redirect all http:// and https:// requests there. The initial nodogsplash pops up with error 404 after the redirect request. I tried searching old closed issues but I don't think I've found anything regarding offline internal php server, only external using FAS. thank you!

**Comment by bluewavenet** [comment]
@raspblu 
This can be made to work, but first, a few points to note:

Devices (mobiles, tablets, laptops and even desktops, have Captive portal Detection (CPD). 

On connecting to a wifi network, the user device CPD tries to access a predetermined port 80 (http) detection web site. 

The actual web site depends on the operating system and vendor of the device.

To do this, the CPD first does a DNS query to get the IP address of the detection web site, then tries to access the web site by ip address on port 80.

An attempt to access anything on port 80 will trigger the NDS splash page, or cause a redirection to the FAS login page.

FAS provides Forward Authentication Service and is just a way of providing a more sophisticated dynamic web page than the simple splash page that NDS provides by default.

The FAS page does not have to be external, it can be on a web server on the same router as NDS if you wish. The FAS can also be a dynamic web page served by the built in NDS web server using the new PreAuth configuration (currently on master and at beta status but about to be released).

Clearly from this, you can see that the CPD requires a positive response from DNS. This will fail if you are offline. To work offline you must configure your DNS server to reply with a local ip address for EVERY enquiry. This is simple to do if you are using Dnsmasq, and can be done with other DNS servers.

Various config problems could result in the 404 error, so I think that checking your config against the documentation will be the best place to start. Also it would be sensible to get it working ONLINE first, before making any changes to your DNS server.

Setup of NDS/FAS/PreAuth etc., plus more in depth description of operation with examples can be found in the online documentation at:
https://nodogsplashdocs.readthedocs.io/en/latest/

Let me know how you get on :-D

**Comment by raspblu** [comment]
@bluewavenet Thanks for the quick and very detailed response! I have read the documentations for FAS and understand a little bit more about how FAS and NDS works. I ended up using the original splash.html so I actually do not need to enable FAS. I actually do not really need an external authentication service. I got the configuration to work and the reason why it was not working was, like you mentioned, no internet. I can change the DNSmasq from my AP to address=/#/192.168.1.1 to redirect all offline http request to NDS but it seems such config affects the raspberry pi own connections to internet to for some reason, so I decided to not bother about offline requests. I see that there’s a further issue about https requests that you mentioned in another thread, but the fact that a device’s auto captive portal already forces NDS splash to appear at the initial connection should be sufficient. Also, I’ve changed domains in hosts config to website.localnet, so it should not be that difficult for someone to reach my internal server once connected. Thanks for the explanation again!
